swarm/src/lib.rs

```diff
@@ -135,29 +142,111 @@ pub type NegotiatedSubstream = Negotiated<Substream<StreamMuxerBox>>;
 
 /// Event generated by the `Swarm`.
 #[derive(Debug)]
-pub enum SwarmEvent<TBvEv> {
+pub enum SwarmEvent<TBvEv, THandleErr> {
     /// Event generated by the `NetworkBehaviour`.
     Behaviour(TBvEv),
-    /// We are now connected to the given peer.
-    Connected(PeerId),
-    /// We are now disconnected from the given peer.
-    Disconnected(PeerId),
-    /// One of our listeners has reported a new local listening address.
-    NewListenAddr(Multiaddr),
-    /// One of our listeners has reported the expiration of a listening address.
-    ExpiredListenAddr(Multiaddr),
+    /// A connection to the given peer has been opened.
+    NewConnection {
```

Comment:
```suggestion
    ConnectionEstablished {
```
The reason for these naming suggestions (and also `ConnectionClosed`) is two-fold:

  * It shows more symmetry, also to the underlying network events.
  * Keeping `Disconnected` can be misleading, especially since it is now emitted for every closed connection.
  * These names would merge quite well with a PR that I have almost ready that introduces `inject_connection_established` / `inject_connection_closed` to the `NetworkBehaviour`.